src/components/complete-profile-fourth/CompleteProfileFourth.tsx

```diff
@@ -128,18 +78,32 @@ export default function CompleteProfileFourth({ onNext }: { onNext: () => void }
             </Select>
           </FormControl>
         </SelectContainer>
+        {invalidTimeError && (
+          <ErrorText>{translate('completeProfileFourth.invalid_time')}</ErrorText>
+        )}
+        {identicalTimeError && (
+          <ErrorText> {translate('completeProfileFourth.equal_time_error')}</ErrorText>
+        )}
         <Snackbar
-          open={error}
+          open={serverError}
           autoHideDuration={2500}
-          onClose={(): void => setError(false)}
+          onClose={(): void => setServerError(false)}
           anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
         >
           <Alert severity="error">{translate('unexpected_error')}</Alert>
         </Snackbar>
         <ProfileBtn
           text={translate('btn_next')}
           onClick={defineAvailableDays}
-          disabled={!daySelected || !availableFrom || !availableTo}
+          disabled={
+            invalidTimeErrors.length > 0 ||
+            identicalTimeErrors.length > 0 ||
+            !daySelected ||
+            !availableFrom ||
+            !availableTo ||
+            invalidTimeError ||
+            availableFrom === availableTo
+          }
```

Comment:
I'm also curious, in this case with 'disabled' property can I make a costant after all defined state like this:
const isDisabled =   invalidTimeErrors.length > 0 ||  identicalTimeErrors.length > 0 ||  !daySelected and so on and use it like so: disabled={isDisabled} or such approach will be causing a rerender?

Reply:
Regardless of where the condition is calculated - directly in JSX in the disabled prop or in a separate variable before JSX, it will be recalculated every time the component is rendered. However, placing logic with several conditions (as you have) outside of JSX makes your code more readable.

Reply:
I think so too, that's why I asked, but from our previous prs I saw that using constants in jsx components is a bad practice, so I'm not sure.

Reply:
If you have multiple conditions in one prop, this is a good practice.